Update SensorDataConverter.java

sveinpg opened the pull request:
Added support for EEG, TMP and PZT. 

Got a feeling that the temperature transfer function provided at the BITalino webpage: http://bitalino.com/datasheets/TMP_Sensor_Datasheet.pdf is wrong. The provided function scales the temperature between -50 and 280 degrees celsius, not -40 and 125 as the documentation specifies.

Commit:
Update SensorDataConverter.java

Added support for EEG, TMP and PZT. 

Got a feeling that the temperature transfer function provided at the BITalino webpage: http://bitalino.com/datasheets/TMP_Sensor_Datasheet.pdf is wrong. Using the function provided scales the temperature between -50 and 280 degrees celsius, not -40 and 125 as the documentation specifies.

pires commented:
Can you double check all other scaling functions? I was the only one looking at them.
Regarding temperature, -40F seems like a bogus temperature.

sveinpg commented:
Sure, I can check them against my local impl.

The documentation states that the range of temperature is -40 to 125
celsius, corresponding to -40 fahrenheit and 257 fahrenheit. But the
transfer function provided by them scales the values between -50 and 265
celsius.

sveinpg

On Tue, Mar 22, 2016 at 1:14 PM, Paulo Pires notifications@github.com
wrote:

> Can you double check all other scaling functions? I was the only one
> looking at them.
> Regarding temperature, -40F seems like a bogus temperature.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/BITalinoWorld/java-sdk/pull/4#issuecomment-199784063

pires commented:
@hugoslv care to shed some light into the above question around temperature?

sveinpg commented:
@pires, I found some deviations in your implementation and the BITalino
documentation. I'll create a pull request to fix them.

Just one thing, scaleEDA() is suppose to be given in the range of 0 to inf.
Do you want me to return infinity, or do you want me to re scale the range?

On Tue, Mar 22, 2016 at 1:17 PM, Svein Petter Gjøby sveinpg@gmail.com
wrote:

> Sure, I can check them against my local impl.
> 
> The documentation states that the range of temperature is -40 to 125
> celsius, corresponding to -40 fahrenheit and 257 fahrenheit. But the
> transfer function provided by them scales the values between -50 and 265
> celsius.
> 
> sveinpg
> 
> On Tue, Mar 22, 2016 at 1:14 PM, Paulo Pires notifications@github.com
> wrote:
> 
> > Can you double check all other scaling functions? I was the only one
> > looking at them.
> > Regarding temperature, -40F seems like a bogus temperature.
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly or view it on GitHub
> > https://github.com/BITalinoWorld/java-sdk/pull/4#issuecomment-199784063

Commit:
Rescaled ECG

Rescaled ECG to -1,5 to 1.5 microVolt

Commit:
Fixed integer division

Fixed integer division in scaleAccelerometerWithPrecision()

Commit:
Rescaled EDA

Rescaled scaleEDA() to 1 to Infinity.

sveinpg commented:
Commited the changes, EMG and LUX returned the correct values.

pires commented:
Please, add [tests and make sure the current ones work](https://github.com/BITalinoWorld/java-sdk/blob/master/src/test/java/com/bitalino/util/SensorDataConverterTest.java).

Commit:
Added tests

Added tests for PZT, EEG and TMP; both celsius and fahrenheit. 
Updated the  assert-value of the ECG and EDA tests.

pires commented:
I've double-checked this and LGTM. @sveinpg your contributions are amazing! Thank you.

sveinpg commented:
@pires could you upload a new release with the new scaling functions?
On Mar 22, 2016 17:28, "Paulo Pires" notifications@github.com wrote:

> Merged #4 https://github.com/BITalinoWorld/java-sdk/pull/4.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/BITalinoWorld/java-sdk/pull/4#event-598969437

pires commented:
https://github.com/BITalinoWorld/java-sdk/releases/tag/1.1.0

hugoslv commented:
@pires @sveinpg the temperature sensor currently being made available in the BITalino lineup is a general-purpose sensor, thought to even be used for environmental applications (hence the wide range).

We haven't made any particular amplification or filtering, so the [-40; 125] ºC interval based on the typical range listed on the data sheet of the chip currently used as the base for this sensor:
http://www.analog.com/media/en/technical-documentation/data-sheets/TMP35_36_37.pdf

For this particular transfer function we've made a few tests, which seemed correct... it also matches the information found on the TMP36 tutorial available on the Adafruit website:
https://learn.adafruit.com/tmp36-temperature-sensor/overview

Let me know if you're experiencing any issues with the data and we'll review this again to double check that the information is being provided as it should.
